qutebrowser/browser/greasemonkey.py

```diff
@@ -0,0 +1,289 @@
+            for(var header in details.headers)
+                oXhr.setRequestHeader(header, details.headers[header]);
+
+        if("data" in details)
+            oXhr.send(details.data);
+        else
+            oXhr.send();
+    }} else
+        throw ("This Browser is not supported, please upgrade.")
+}}
+
+function GM_addStyle(/* String */ styles) {{
+    var head = document.getElementsByTagName("head")[0];
+    if (head === undefined) {{
+        document.onreadystatechange = function() {{
+            if (document.readyState == "interactive") {{
+                var oStyle = document.createElement("style");
+                oStyle.setAttribute("type", "text\/css");
+                oStyle.appendChild(document.createTextNode(styles));
+                document.getElementsByTagName("head")[0].appendChild(oStyle);
+            }}
+        }}
+    }}
+    else {{
+        var oStyle = document.createElement("style");
+        oStyle.setAttribute("type", "text\/css");
+        oStyle.appendChild(document.createTextNode(styles));
+        head.appendChild(oStyle);
+    }}
+}}
+
+unsafeWindow = window;
+"""
+
+    def __init__(self, properties, code):
+        self._code = code
+        self._includes = []
+        self._excludes = []
+        self._description = None
+        self._name = None
+        self._run_at = None
+        for name, value in properties:
+            if name == 'name':
+                self._name = value
+            if name == 'description':
+                self._description = value
+            if name in ['include', 'match']:
+                self._includes.append(value)
+            if name in ['exclude', 'exclude_match']:
+                self._excludes.append(value)
+            if name == 'run-at':
+                self._run_at = value
+
+    HEADER_REGEX = r'\/\/ ==UserScript==.|\n+\/\/ ==\/UserScript==\n'
+    PROPS_REGEX = r'\/\/ @(?P<prop>[^\s]+)\s+(?P<val>.+)'
+
+    @classmethod
+    def parse(cls, source):
+        props, _code = re.split(cls.HEADER_REGEX, source)
+        return cls(re.findall(cls.PROPS_REGEX, props), source)
+
+    def includes(self):
+        return self._includes
+
+    def excludes(self):
+        return self._excludes
+
+    def name(self):
+        return self._name
+
+    def run_at(self):
+        return self._run_at
+
+    def code(self):
+        gm_bootstrap = self.GM_BOOTSTRAP_TEMPLATE.format(
+            scriptName=self._name,
+            scriptInfo=self.meta_json() or 'null',
+            scriptMeta=self.meta_raw() or 'null')
+        return '\n'.join(
+            ["(function(){", gm_bootstrap, self._code, "}).call();"])
+
+    def meta_json(self):
+        return json.dumps({
+            'name': self._name,
+            'description': self._description,
+            'matches': self._includes,
+            'includes': self._includes,
+            'excludes': self._excludes,
+            'run-at': self._run_at,
+        })
+
+    def meta_raw(self):
+        pass
```

Comment:
What's this for?

Reply:
Hmm, https://wiki.greasespot.net/GM_info
For the GM_info function one of the properties of the return object is scriptMetaStr "the entire metadata block of the script.
Added as a attribute of script.